test.js

```diff
@@ -35,9 +34,9 @@ test('rehypeRaw', (t) => {
         '</div>'
       ].join('\n'),
       (error, file) => {
```

Comment:
This is the execution order now:

1. the test runs
2. The unified processor is started
3. The test succeeds
4. The unified processor calls a callback
5. Assertions are made

Previously `tape` could use the `t` context to determine how many assertions were expected, and wait for those. This is no longer the case.

IMO the cleanest solution is to either:

- use the unified promise API, and async / await syntax for this test.
- use the unified sync API.

Reply:
I don't think it has been reordered in that way.
I tried adding `assert.fail()` inside the callback to see if it would wait long enough for it to fail.
It does fail.

I'm open to making it async if that would be more clear for folks.
But I don't think the change is required for the tests to function.

Reply:
I think Remco here is correct, due to `plan`. That makes tape run async and wait until there are two assertions.

I think what’s happening now is just uncaught exceptions (because `assert` throws errors, tape doesn’t). While it “works” (because uncaught exceptions still fail the process), I don’t think that’s a good idea?

So I believe your `assert.fail` causes an uncaught exception?

I like using async/await now :)